### PR TITLE
Document node-local model of blocks

### PR DESCRIPTION
**Pull request by lucafavatella**
In scope of #255 .

### CODE REVIEW

**Comment by lucafavatella** [comment]
I did not fully assess conflicts with #297 following its recent rebase/update. The item I am aware of could cause merge issues is the `block_absorber`.

**Comment by cytadela8** [comment]
@lucafavatella You can change the header:absorb calls, but as I said this may break fragile MarketTest. There is (should not be) any need to call it twice.

File: apps/ae_core/src/consensus/tester.erl [hunk 1]
```diff
@@ -22,8 +22,6 @@ test1() ->
     S = ae_core_fractions:test(),
     lager:info("merkel tree tests"),
     S = tree_test:test(),
-    lager:info("block hashes test"),
```
Comment (sennui) [inline apps/ae_core/src/consensus/tester.erl]: why did you remove the test call?

Comment (lucafavatella) [inline apps/ae_core/src/consensus/tester.erl]: I deleted the test call because I deleted `block_hashes:test/0`, and I deleted `block_hashes:test/0` because it is a test on the documented API of `gb_sets` (new/0, add/2, is_member/2).

**Comment by lucafavatella** [comment]
I am identifying block-related changes that went in master in the meantime - then I will rebase.

**Comment by lucafavatella** [comment]
I see #299 introduced function but not used yet.
I rebased and force-pushed (there was only one comment) - I will keep an eye on the CI.

**Comment by lucafavatella** [comment]
@sennui Ready to merge then.
